File: Code/Tier2/Common/src/main/java/javaModel/Group.java
package javaModel;


import java.io.Serializable;
import java.util.ArrayList;

public class Group implements Serializable {

    public SprintList sprintList;
    public ProductBacklog productBacklog;
    public ArrayList<GroupMember> groupMembers;
    private boolean validated;
    private int groupId;

    public int getGroupId() {
        return groupId;
    }

    public void setGroupId(int groupId) {
        this.groupId = groupId;
    }

    public SprintList getSprintList() {
        return sprintList;
    }

    public void setSprintList(SprintList sprintList) {
        this.sprintList = sprintList;
    }

    public ProductBacklog getProductBacklog() {
        return productBacklog;
    }

    public void setProductBacklog(ProductBacklog productBacklog) {
        this.productBacklog = productBacklog;
    }

    public void setGroupMembers(ArrayList<GroupMember> groupMembers) {
        this.groupMembers = groupMembers;
    }

    public boolean isValidated() {
        return validated;
    }

    public void setValidated(boolean validated) {
        this.validated = validated;
    }

    public Group() {
        groupMembers= new ArrayList<>();
        sprintList= new SprintList();
        productBacklog = new ProductBacklog();
    }


    public Group(SprintList sprintList, ProductBacklog productBacklog,int groupId) {
        this.sprintList = sprintList;
        this.productBacklog = productBacklog;
        groupMembers = new ArrayList<>();
        validated =false;
        this.groupId = groupId;
    }

    public void addMember(GroupMember e) {
        groupMembers.add(e);
    }

    public void removeMemberByMember(GroupMember e) {
        for (int i = 0; i < groupMembers.size(); i++) {
            if (groupMembers.get(i).equals(e)) {
                groupMembers.remove(i);
            }
        }
    }

    public void removeGroupMemberByIndex(int index) {
        groupMembers.remove(index);
    }

    public GroupMember[] getGroupMembers() {
        GroupMember[] membersArray = new GroupMember[groupMembers.size()];
        membersArray = groupMembers.toArray(membersArray);
        return membersArray;
    }



}

File: Code/Tier3-DatabaseLayer/RestServer/Common/Group.cs
using System;
using System.Collections.Generic;

namespace WebApplication.Common
{
    public class Group
    {
        public SprintList sprintList { get; set; }
        public ProductBacklog productBacklog { get; set; }
        public List<GroupMember> groupMembers { get; set; }
        public int groupId { get; set; }
        public bool validated { get; set; }

        public Group()
        {
            sprintList = new SprintList();
            productBacklog = new ProductBacklog();
            groupMembers = new List<GroupMember>();
        }

        public Group(int id)
        {
            sprintList = new SprintList();
            productBacklog = new ProductBacklog();
            groupId = id;
            groupMembers = new List<GroupMember>();
        }

        public void AddMember(GroupMember e)
        {
            groupMembers.Add(e);
        }

        public void RemoveMemberByMember(GroupMember e)
        {
            groupMembers.Remove(e);
        }

        public void RemoveGroupMemberByIndex(int index)
        {
            groupMembers.RemoveAt(index);
        }
        
        public override string ToString()
        {
            return
                         $"SprintList: {sprintList}, ProductBacklog: {productBacklog}, GroupMembers: {groupMembers}, GroupId: {groupId}, validated: {validated}";
                 }
    }
}